climpred/graphics.py

```diff
@@ -230,3 +236,194 @@ def plot_bootstrapped_skill_over_leadyear(bootstrapped, plot_persistence=True, a
+        )
+
+    for i, lead in enumerate(hind.lead.values):
+        h = hind.sel(lead=lead).rename({'init': 'time'})
+        if not show_members and 'member' in h.dims:
+            h = h.mean('member')
+            lead_alpha = 1
+        else:
+            lead_alpha = 0.5
+        h['time'] = shift_cftime_index(h.time, 'time', int(lead), lead_freq)
+        h.plot(
+            ax=ax,
+            hue='member',
+            color=cmap(i),
+            label=f'initialized: lead={lead}',
+            alpha=lead_alpha,
+            zorder=hind.lead.size - i,
+        )
+
+    linestyles = ['-', ':', '-.', '--']
+    if len(obs) > 0:
+        for i, (obs_name, obs_item) in enumerate(obs.items()):
+            obs_item[variable].plot(
+                ax=ax,
+                color='k',
+                lw=3,
+                ls=linestyles[i],
+                label=f'reference: {obs_name}',
+                zorder=hind.lead.size + 1,
+            )
+
+    # show only one item per label in legend
+    handles, labels = ax.get_legend_handles_labels()
+    by_label = OrderedDict(zip(labels, handles))
+    ax.legend(
+        by_label.values(), by_label.keys(), loc='center left', bbox_to_anchor=(1, 0.5)
+    )
+    ax.set_title('')
+    return ax
+
+
+def plot_ensemble_perfect_model(
+    pm, variable=None, ax=False, show_members=True, cmap='tab10'
+):
+    """Plot datasets from PerfectModelEnsemble.
+
+    Args:
+        pm (PredictionEnsemble): PredictionEnsemble.
+        variable (str or None): `variable` to plot. Defaults to the first in data_vars.
+        ax (plt.axes): axis to plot to. Defaults to False. Creates new.
+        show_members (bool): whether to display all members individually.
+            Defaults to True.
+        cmap (str): Name of plt colorbar. Defaults to 'tab10'.
+
+    Returns:
+        ax: plt.axes
+
+    """
+
+    check_only_climpred_dims(pm)
+    if not variable:
+        variable = list(pm._datasets['initialized'].data_vars)[0]
+    initialized = pm._datasets['initialized'][variable]
+    control = pm._datasets['control']
+    if isinstance(control, xr.Dataset):
+        control = control[variable]
+    calendar = infer_calendar_name(initialized.init)
+    lead_freq = get_lead_cftime_shift_args(initialized.lead.attrs['units'], 1)[1]
+
+    if not ax:
+        _, ax = plt.subplots(figsize=(10, 4))
+
+    cmap = mpl.cm.get_cmap(cmap, initialized.init.size)
+
+    for ii, i in enumerate(initialized.init.values):
+        dsi = initialized.sel(init=i).rename({'lead': 'time'})
+        # convert lead time into cftime
+        start_str = i.strftime()[:10]
+        # yi,mi,di = i.strftime().split(' ')[0].split('-')
+        # i_cfdatetime = cftime.datetime(int(yi),int(mi), int(di))
+        # try to plot vertical bars for each init
+        # now fails todo: how to axvline(cftime)?
+        # ax.axvline(x=i_cfdatetime,calendar='noleap')
+        # ax.axvline(x=int(init.init.isel(init=[i]).dt.year.values))
```

Comment:
make sure these get deleted if not needed

Reply:
Do you have an idea how to solve this? axvline in cftime units?

Reply:
Honestly I didn't read this -- just saw comment blocks. I can give this a look tomorrow.

Reply:
this is not too important